modules/neurons/training/reinforcement_loop.py

```
@@ -1127,6 +1151,356 @@ def _prepare_dataset(
+        }
+
+
+class ReinforcementLoop:
+    """Execute Unsloth-backed GRPO cycles focused on reasoning prompts."""
+
+    _GENERATION_KWARGS = {
+        "max_new_tokens": 512,
+        "do_sample": True,
+        "temperature": 0.7,
+        "top_p": 0.9,
+    }
```

Ruff (0.13.3)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)